daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DFDLDelimiter.scala

```diff
@@ -602,3 +619,15 @@ class WSPStarDelim extends WSPBase with WSP {
 
   def unparseValue(outputNewLine: String): String = ""
 }
+
+// This delimiter matches an empty space which effectively is always a
+// successful match without moving the character position. This effectively
+// acts as a maker that empty match is allowed
```

Comment:
Should be scaladoc.

Please add a little more description of why this isn't a stupid-nonsensical thing to do. That ES as a terminator when lengthKind is other than 'delimited' is a way of indicating an optional terminator, but that this DFA/scanner is used to consume delimiters in that case as well. This class establishes the invariant that there is always a DelimBase object used when "scanning", even when we're "scanning for nothing".